lib/src/revset.rs

```diff
@@ -231,6 +231,9 @@ pub type ResolvedRevsetExpression = RevsetExpression<ResolvedExpressionState>;
 pub enum RevsetExpression<St: ExpressionState> {
     None,
     All,
+    /// Visible heads and all referenced commits within the current expression
+    /// scope. Used as the default of `Range`/`DagRange` heads.
+    ReferencedHeads,
```

Comment:
I think `AllHeads` seems like a better name

Reply:
I avoided `AllHeads` because it sounds like all historical heads. How about `ReferencedOrVisibleHeads`?

Reply:
I would not make the connection to historical heads, so I still prefer `AllHeads`. It matches `All` (which I also had not thought might include historical commits). If we add expressions for all historical heads, then `HistoricalHeads` sounds like a good name to me.

Reply:
In context of https://github.com/jj-vcs/jj/pull/6782#discussion_r2155998760, if we're going to add `all_heads()`, this `ReferencedHeads` should have to be named differently since `AllHeads = Heads(ReferencedHeads)`.